pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -194,7 +194,9 @@ func (e *Controller) Run(workers int, stopCh <-chan struct{}) {
 	}
 
 	for i := 0; i < workers; i++ {
-		go wait.Until(e.worker, e.workerLoopPeriod, stopCh)
+		go wait.Until(func() {
```

Comment:
I suggest opening two alternatives.  This is one.  Another is a breaking change on `wait.Until` and we talk about it at apimachinery next week.  Put it on the agenda and add to the mailing list.

Reply:
What is the breaking change? Being able to pass parameters to the function in `wait.Until` rather than just a function reference?

Reply:
you need the wait function to actually provide context, so the conditionfunc changes.